mp4/box.go

```diff
@@ -146,7 +146,7 @@ func (b boxHeader) payloadLen() int {
 // decodeHeader decodes a box header (size + box type + possiible largeSize)
 func decodeHeader(r io.Reader) (boxHeader, error) {
 	buf := make([]byte, boxHeaderSize)
-	n, err := r.Read(buf)
+	n, err := io.ReadFull(r, buf)
```

Comment:
In this particular case, we actually check n, which will always be 8 when we use ReadFull.